sql/sqlcheck/sqlcheck.go

```diff
@@ -133,6 +133,56 @@ var Destructive = &driverAware{
 	},
 }
 
+// Renames detects potential column renames.
+var Renames = &driverAware{
+	run: func(ctx context.Context, diags []Diagnostic, p *Pass) error {
+		for _, sc := range p.File.Changes {
+			for _, c := range sc.Changes {
+				c, ok := c.(*schema.ModifyTable)
+				if !ok {
+					continue
+				}
+			Loop:
+				for i := range c.Changes {
+					d, ok := c.Changes[i].(*schema.DropColumn)
+					if !ok {
+						continue
+					}
+					for j := range c.Changes[i+1:] {
+						a, ok := c.Changes[i+1+j].(*schema.AddColumn)
+						if !ok {
+							continue
+						}
+						// Use schema.Differ to see if type has changed, minimal setup to test only type changes
+						fromTbl := schema.NewTable("f").AddColumns(schema.NewColumn("diff").SetType(d.C.Type.Type))
+						toTbl := schema.NewTable("f").AddColumns(schema.NewColumn("diff").SetType(a.C.Type.Type))
```

Comment:
The type is not enough.
The columns should look the same. Hence, they need to have the same attributes.

Reply:
True. Also, please make sure to add a test for this.